pages/api/v1/answers.ts

```diff
@@ -675,13 +680,18 @@ async function Angela( // Good ol' Angela
                     .replace(/[^A-Z]+/gi, "")
                     .includes("askuser")
                 ) {
+                  // AI set parameter
                   command.args[param] = response;
                 } else {
+                  // AI said we need to 'Ask user'
                   console.info("Needs user correction: ", param);
                   needsUserCorrection = true;
-                  toUserCorrect.push(param);
+                  toUserCorrect.push({ functionName: command.name, param });
                 }
               }
+              // Update the past assistant message with newly-added parameters (so it looks to future AI
+              //  that it got the call right first time - stops it thinking it can skip required parameters)
+              updatePastAssistantMessage(command, nonSystemMessages);
```

Comment:
Does this work if there's more than one parameter to correct? It seems to be outside of the above loop and there's no test for multiparam

Reply:
Yes, because `command.args` contains all the new params output by param correction now. If there are still some missing (i.e. some `ask user`), then they won't be included (since they aren't in `command.args`)

Reply:
can you add a test for multiparam correction 🙏